crates/ra_ide/src/display/function_signature.rs

```diff
@@ -207,7 +207,18 @@ impl From<&'_ ast::FnDef> for FunctionSignature {
                     res.push(raw_param);
                 }
 
-                res.extend(param_list.params().map(|param| param.syntax().text().to_string()));
+                // macro-generated functions are missing whitespace
+                fn fmt_param(param: ast::Param) -> String {
+                    let text = param.syntax().text().to_string();
+                    match text.find(':') {
+                        Some(pos) if 1 + pos < text.len() => {
```

Comment:
I think this condition is always true. We already have this function already actually:

https://github.com/rust-analyzer/rust-analyzer/blob/dff8140a2e9de8b6d9c64cfcc9a8d16db3a0fecd/crates/test_utils/src/lib.rs#L335-L338

Could you move this `split1` to `stdx/lib.rs`, and then use it here and in test_utils?

Otherwise, LGTM!

bors d+